backend/src/test/java/codezap/auth/configuration/AuthArgumentResolverTest.java

```diff
@@ -0,0 +1,161 @@
+            Method supportMethod = ReflectionSupport.findMethod(
+                            SupportTestController.class, "supportMethod", Member.class)
+                    .orElseThrow();
+            MethodParameter methodParameter = new MethodParameter(supportMethod, 0);
+
+            assertThat(authArgumentResolver.supportsParameter(methodParameter))
+                    .isTrue();
+        }
+
+        @Test
+        @DisplayName("성공: AuthenticationPrinciple 어노테이션이 파라미터에 존재하지 않으면 지원할 수 없는 메서드이다.")
+        void notSupportsTest() {
+            Method supportMethod = ReflectionSupport.findMethod(
+                            SupportTestController.class, "notSupportMethod", Member.class)
+                    .orElseThrow();
+            MethodParameter methodParameter = new MethodParameter(supportMethod, 0);
+
+            assertThat(authArgumentResolver.supportsParameter(methodParameter))
+                    .isFalse();
+        }
+    }
+
+    @Nested
+    @DisplayName("파라미터 반환 테스트")
+    class ResolveArgument {
+
+        static class ResolveTestController {
+            public void notRequiredMethod(@AuthenticationPrinciple(required = false) Member member) {
+            }
+
+            public void requiredMethod(@AuthenticationPrinciple Member member) {
+            }
+        }
+
+        private final MockHttpServletRequest httpServletRequest = new MockHttpServletRequest("GET", "/templates");
+        private final StandardServletAsyncWebRequest nativeWebRequest =
+                new StandardServletAsyncWebRequest(httpServletRequest, new MockHttpServletResponse());
+        private final Member member = MemberFixture.getFirstMember();
+        private final Cookie credentialCookie = new Cookie(CREDENTIAL_COOKIE_NAME, credentialProvider.createCredential(member));
+
+        @Nested
+        @DisplayName("required 값이 false 라면")
+        class RequiredFalseTest {
+            Method notRequiredMethod = ReflectionSupport.findMethod(ResolveTestController.class,
+                            "notRequiredMethod", Member.class)
+                    .orElseThrow();
+
+            @Test
+            @DisplayName("성공: credential 정보가 없을때 null 이 반환된다.")
+            void noCredentialTest() {
+                //given
+
+                //when
+                Member member = resolveArgument(notRequiredMethod, nativeWebRequest);
+
+                //then
+                assertThat(member).isNull();
+            }
+
+            @Test
+            @DisplayName("성공: credential 정보가 존재하면 Member 가 반환된다.")
+            void existCredentialTest() {
+                //given
+                httpServletRequest.setCookies(credentialCookie);
+
+                //when
+                Member resolvedArgument = resolveArgument(notRequiredMethod, nativeWebRequest);
+
+                //then
+                assertThat(resolvedArgument).isEqualTo(member);
+            }
+        }
+
+        @Nested
+        @DisplayName("required 값이 true 라면")
```

Comment:
🍜? ㅋㅋㅋㅋㅋ
```suggestion
        @DisplayName("required 값이 true 일 경우")
```
로 수정하는 건 어떤가요?

Reply:
🍜  호록,,,